crates/biome_js_formatter/src/js/expressions/template_expression.rs

```diff
@@ -99,3 +138,333 @@ impl NeedsParentheses for JsTemplateExpression {
         }
     }
 }
+
+/// A template literal contains `JsTemplateChunkElement` and `JsTemplateElement` elements.
+/// A `JsTemplateElement` is an expression that cannot be directly passed to a foreign language formatter.
+/// This is because formatters for other languages, such as CSS, do not understand JavaScript expressions.
+/// Therefore, we need to replace the JavaScript expression with a placeholder.
+///
+/// Consider the following example:
+/// ```js
+/// const css = css`
+///    background: ${color};
+/// `;
+/// ```
+/// First, we need to replace `${color}` with a placeholder.
+/// After this replacement, the template literal will look like this:
+/// ```js
+/// const css = css`
+///   background: placeholder;
+/// `;
+/// ```
+/// This modified template can then be passed to the CSS formatter.
+/// Once formatting is complete, we need to replace the placeholder with the original expression.
+fn format_embedded_language(
+    elements: JsTemplateElementList,
+    language: JsForeignLanguage,
+    f: &mut JsFormatter,
+) -> FormatResult<()> {
+    // We need to track the relationship between placeholders and the original expressions.
+    // There is a special scenario we need to handle:
+    // If string interpolations are adjacent to each other, replacing them one by one with placeholders
+    // makes it difficult to replace them back with the original expressions.
+    // Therefore, we need to group adjacent interpolations together and replace them with a single placeholder.
+    //
+    // For example, consider the following:
+    // `background: ${bg}${color}`
+    // We should treat `${bg}${color}` as a group and replace it with a single placeholder:
+    // `background: placeholder`
+    // The placeholder is then mapped to `[bg, color]`.
+    let mut placeholder_map: HashMap<String, Vec<JsTemplateElement>> = HashMap::new();
+    let mut index = 0;
+
+    let mut content = String::new();
+    let mut element_iter = elements.iter().peekable();
+    while let Some(template_element) = element_iter.next() {
+        match template_element {
+            AnyJsTemplateElement::JsTemplateChunkElement(element) => {
+                content.push_str(element.text().as_str());
+            }
+            AnyJsTemplateElement::JsTemplateElement(element) => {
+                let mut string_interpolations = vec![element.clone()];
+                while let Some(AnyJsTemplateElement::JsTemplateElement(element)) =
+                    element_iter.peek()
+                {
+                    string_interpolations.push(element.clone());
+                    element_iter.next();
+                }
+                let placeholder = std::format!("biome-placeholder-{}", index);
```

Comment:
Oh, and another one, `-` is not allowed in a gritql metavariable name: https://docs.grit.io/language/patterns#metavariables, it uses `$lowercase_snake_case`